src/libs/actions/Report.js

```diff
@@ -586,8 +585,13 @@ function addAction(reportID, text, file) {
     // Generate a clientID so we can save the optimistic action to storage with the clientID as key. Later, we will
     // remove the optimistic action when we add the real action created in the server. We do this because it's not
     // safe to assume that this will use the very next sequenceNumber. An action created by another can overwrite that
-    // sequenceNumber if it is created before this one.
-    const optimisticReportActionID = Str.guid(`${Date.now()}_`);
+    // sequenceNumber if it is created before this one. We use a combination of current epoch timestamp (milliseconds)
+    // and a random number so that the probability of someone else having the same optimisticReportActionID is
+    // extremely low even if they left the comment at the same moment as another user on the same report. The random
+    // number is 3 digits because if we go any higher JS will convert the digits after the 16th position to 0's in
+    // optimisticReportActionID.
+    const randomNumber = Math.floor((Math.random() * (999 - 100)) + 100);
+    const optimisticReportActionID = parseInt(`${Date.now()}${randomNumber}`, 10);
```

Comment:
Looks good, but does passing this as a `Number` to the API make any difference?

Reply:
https://github.com/Expensify/Web-Expensify/blob/d40de31a43bdeb4662ee059dbf4095e57fe02c68/lib/ReportAPI.php#L291

Reply:
It does not make any difference from what I tested. It's been string for a while so I let it be as is.

The current flow is like so,
1. optimisticReportActionID number is created using date and random number.
2. Its passed as a number but when PHP receives it its type converted to string. 
3. It works just fine and i didn't notice any issue around it on PHP. tailed logs and didn't see any issues.
4. Finally PHP returns the client_id as a string, but when we use it as an object key [here](https://github.com/Expensify/Expensify.cash/blob/265023c89558984a0ad400752c94f9d028810797/src/libs/actions/Report.js#L268) it's automatically converted to a number again and hence removes the optimisticReportActionID from onyx report actions. 

Let me know if there is something i overlooked.

Reply:
Nope that sounds good to me.